libminifi/include/utils/Id.h

```diff
@@ -43,107 +44,41 @@ class uuid;
 #define UUID_TIME_STR "time"
 #define UUID_WINDOWS_STR "windows"
 
-
 namespace org {
 namespace apache {
 namespace nifi {
 namespace minifi {
 namespace utils {
 
-template<typename T, typename C>
-class IdentifierBase {
+class Identifier {
+  friend struct IdentifierTestAccessor;
  public:
-  IdentifierBase(T myid) { // NOLINT
-    copyInto(myid);
-  }
-
-  IdentifierBase(const IdentifierBase &other) {
-    copyInto(other.id_);
-  }
-
-  IdentifierBase(IdentifierBase &&other)
-      : converted_(std::move(other.converted_)) {
-    copyInto(other.id_);
-  }
-
-  IdentifierBase() = default;
-
-  IdentifierBase &operator=(const IdentifierBase &other) {
-    copyInto(other.id_);
-    return *this;
-  }
-
-  IdentifierBase &operator=(T o) {
-    copyInto(o);
-    return *this;
-  }
+  using Data = std::array<uint8_t, 16>;
 
-  void getIdentifier(T other) const {
-    copyOutOf(other);
-  }
+  Identifier() = default;
+  explicit Identifier(const Data& data);
+  explicit Identifier(const std::string& data);
```

Comment:
removed the `std::string` constructor as it was not really used, but the `std::string` assignment is more frequently used, and since other PR-s are modifying those parts IMO we should wait till these things are merged

Reply:
Which PRs are you aware of?

Reply:
as the PR I was referring to has just been merged I gave a try to removing the `std::string` assignment one, but there are numerous places it is in use, and most places would need careful thinking if we can handle the error and if so how, so I would still do it in a separate PR